### ISSUE TITLE
Allow deleting content inside the app

### ISSUE THREAD

**Issue by lauriii** [jsdrupal/drupal-admin-ui#119]
Deleting content works currently by redirecting to Drupal's delete content form. We could improve the UX by allowing deleting content directly on the app. Material UI has tools in place for this use case: https://material.io/design/components/dialogs.html#alert-dialog

**Comment by iberdinsky-skilld** [comment]
One thing not obvious here. According to React practice we have to create new dialog component. But i think it should be in atoms and reusable by other app components. and in this case it should be component: Dialog with alert variant here.

**Comment by martinfrances107** [comment]
@mattgrill  -- Are you actively working on this  @?... I have a first draft I am about to create a pull request

I am just posting a early draft - so we can grasp the rough shape of this.

I just copied the example source code from the alert dialog button here 
https://material-ui.com/demos/dialogs/#alerts

For D8 purposes DeleteButton may not be generic enough -- maybe a first cut will highlight any problem areas.

It is not functional ... it just displays a modal which matches the text produced by D8.6's confirm form and then does NOTHING

MY TODO list

1) There are some styling snags .. the shape of the new delete button needs to be circular like the other buttons 
2) Make pressing confirm on the dialog functional.

PS thanks for taking the time to describe an novice issue lauriii++

**Comment by lauriii** [comment]
@martinfrances107 I posted feedback on the UI at the PR! I haven't heard @mattgrill mentioning this issue for a while so I'm unassigning him from here for now 🙂

**Comment by dawehner** [comment]
@martinfrances107 Some of the changes in https://github.com/jsdrupal/drupal-admin-ui/pull/135 could be helpful for you too. It for example provides delete support on the API level and provides a redux action as well.

**Comment by martinfrances107** [comment]
Here are my notes associated with the next PR.

@lauriii  Thanks for the feedback ... after reading your notes and after a little thinking,  I have made the button more component like.

 ... As a component I want a generic  button that lives in a Drupal 'operations' column for whatever table and function comes our way next. One that can be labelled as 'add' 'edit' whatever..and can be styled with a Icon image as per the want of the next developer.

The name I went with was "OpsModalButton" - but I don't particularly feel married to it.

On my todo list -- I still have to fix up your feedback about the in modal button labels.

But in the next iteration I am going to work on the more meaty side of the problem -- and use dawehner's suggestion to implement  the required function. ( Thanks dawehner BTW)
